### PR TITLE
Feature/22416 client sorting & front end testing

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
- Extract library with config
- Frontend testing

### CODE REVIEW

File: admin/assets/javascripts/table-sorting.js [hunk 1]
```diff
@@ -1,28 +1,58 @@
...
+      }
+      if ((v === undefined || v === null || v === '')) {
+        return true
+      }
+      var hasIgnoredString = config.ignoredStrings.some(function (ignoredString) {
+        return ignoredString === v
+      })
+      if (hasIgnoredString) {
+        return true
+      }
```
Comment (jon-shipley) [inline admin/assets/javascripts/table-sorting.js]: This is nice, but it seems to me there is a missing `return false` from the end of the function.

Comment (GeorgeChatzigiannis) [reply admin/assets/javascripts/table-sorting.js]: @jon-shipley done

**Comment by jon-shipley** [comment]
Just remembered to review the folder structure 
`admin/spec/javascripts/table-sorting.spec.js` - do we collectively think that `javascripts` is descriptive enough?  I would prefer `front-end`.  What does everyone else think? @GuyHarwood @danwilliams

**Comment by GeorgeChatzigiannis** [comment]
@jon-shipley I was thinking about this as well. `javascripts` matches the folder name where the source files are but I'd rather go for 'frontend' (no dashes) as name. Thoughts?

**Comment by GuyHarwood** [comment]
@jon-shipley @GeorgeChatzigiannis i would prefer `front-end`

**Comment by GeorgeChatzigiannis** [comment]
@GuyHarwood @jon-shipley done.

File: admin/spec/support/jasmine.json [hunk 2]
```diff
@@ -1,7 +1,11 @@
 {
   "spec_dir": "spec",
   "spec_files": [
-    "**/*[sS]pec.js"
+    "controllers/*[sS]pec.js",
+    "lib/*[sS]pec.js",
+    "lib/**/*[sS]pec.js",
+    "service/*[sS]pec.js",
+    "service/**/*[sS]pec.js"
```
[the remark this answers is not in the document]
Comment (jon-shipley) [reply admin/spec/support/jasmine.json]: We could organise it into 

`spec/front-end`
`spec/back-end`

That would make the above `files` section more obvious.

[the remark this answers is not in the document]
Comment (GeorgeChatzigiannis) [reply admin/spec/support/jasmine.json]: @jon-shipley thats exactly what I tried last night.
Problem is the references do not auto update for some reason and I have to go through them 1 by 1.

**Comment by GeorgeChatzigiannis** [comment]
@GuyHarwood directory is renamed to front-end already.

**Comment by GeorgeChatzigiannis** [comment]
@GuyHarwood @jon-shipley I'll split them to two directories back-end and front-end since I fully agree with this and will update when everything is sorted.

**Pull request merged**